src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -258,190 +237,187 @@ describe("FocusTrap", () => {
           shiftKey: true,
           type: "keydown",
         }),
-        document.querySelectorAll("button")[0],
-        document.querySelectorAll("button")[1]
+        screen.getByRole("button", { name: BUTTON_ONE }),
+        screen.getByRole("button", { name: BUTTON_TWO })
       );
     });
   });
 
   describe("when FocusTrap wraps an element", () => {
     describe("and element has focusable items inside", () => {
       beforeEach(() => {
-        render(
-          <MockComponent>
-            <button type="button">{BUTTON_ONE}</button>
-            <button type="button">{BUTTON_TWO}</button>
-          </MockComponent>
-        );
+        renderMockComponent();
       });
 
       it("should focus the wrapper", () => {
-        expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+        expect(screen.getByTestId(WRAPPER_ID)).toHaveFocus();
       });
 
       it("should not move focus if different key than TAB is pressed", () => {
         focusElement(screen.getByText(BUTTON_TWO));
         fireEvent.keyDown(screen.getByTestId(WRAPPER_ID), { key: "ArrowDown" });
-        expect(screen.getByText(BUTTON_TWO)).toBeFocused();
+        expect(screen.getByText(BUTTON_TWO)).toHaveFocus();
       });
 
       it("should move focus back to the last item when `shift + tab` pressed and first focusable item is activeElement", () => {
         focusElement(screen.getByText(BUTTON_ONE));
         shiftTabPress();
-        expect(screen.getByText(BUTTON_TWO)).toBeFocused();
+        expect(screen.getByText(BUTTON_TWO)).toHaveFocus();
       });
 
       it("should back to the first item when use `shift + tab`", () => {
         focusElement(screen.getByText(BUTTON_TWO));
         shiftTabPress();
-        expect(screen.getByText(BUTTON_ONE)).toBeFocused();
+        expect(screen.getByText(BUTTON_ONE)).toHaveFocus();
       });
 
       it("should go to the second item when use TAB", () => {
         focusElement(screen.getByText(BUTTON_ONE));
         tabPress();
-        expect(screen.getByText(BUTTON_TWO)).toBeFocused();
+        expect(screen.getByText(BUTTON_TWO)).toHaveFocus();
       });
 
       it("should move to the first focusable item if TAB pressed on last focusable item", () => {
         focusElement(screen.getByText(BUTTON_TWO));
         tabPress();
-        expect(screen.getByText(BUTTON_ONE)).toBeFocused();
+        expect(screen.getByText(BUTTON_ONE)).toHaveFocus();
       });
     });
 
     describe("and element does not have focusable items", () => {
       beforeEach(() => {
-        render(
-          <MockComponent>
-            <p>Test content</p>
-          </MockComponent>
-        );
+        renderMockComponent({ children: <p>Test content</p> });
```

Comment:
actually when there are bespoke children I think `render(...)` reads better than `renderMockComponent({children: ...})`, particularly in the later examples with complex children. But I don't mind if it's kept as is.

Reply:
Yeah inclined to agree, especially based on my above comment about refactoring to support the rerender test cases